### ISSUE TITLE
[Bug Report] Scoped CSS does not work for components with `el-tooltip` as root

### ISSUE THREAD

**Issue by Eiryyy** [element-plus/element-plus#5432]
### Element Plus version
1.3.0-beta.5

### OS/Browsers version
macOS 12.1 / Chrome 97.0.4692.71 

### Vue version
3.2.27

### Reproduction Link
https://codesandbox.io/s/element-plus-spa-forked-s8os0?file=/src/App.vue

### Steps to reproduce
1. Classify components with el-tooltip as root (e.g. `el-date-picker`, `el-cascader`) and write scoped css

### What is Expected?
Scoped CSS works.

### What is actually happening?
These components do not have the `data-v-[hash]` attribute.

**Comment by weidehai** [comment]
el-tooltip isn't the root of el-date-picker, el-cascader component

**Comment by Eiryyy** [comment]
I think the root of `el-cascader` is `el-tooltip`.
https://github.com/element-plus/element-plus/blob/df3be7c66117149273f835f4e11654c66fd5c0dd/packages/components/cascader/src/index.vue#L2

Anyway, it is true that scoped CSS is not working.

**Comment by weidehai** [comment]
yes,you are right,i think that because el-tooltip root is what default slot,the complier unable to know that,It's about how scopeid implementation

**Comment by krystian-wolarek** [comment]
Any suggestions on how to deal with this, or some plans to fix it? This is another breaking change not mentioned in doc that forces us to use deep everywhere or to check deeply if some component have tooltip on top or not
